gusto/state.py

```diff
@@ -210,12 +316,26 @@ def setup_dump(self, pickup=False):
             field = Function(functionspaceimpl.WithGeometry(f.function_space(), mesh_ll), val=f.topological, name=name+'_ll')
             self.to_dump_latlon.append(field)
 
-    def dump(self, t=0, diagnostic_everydump=False, pickup=False):
+        # we create new netcdf files to write to, unless pickup=True, in
+        # which case we just need the filenames
+        diagnostics_filename = self.dumpdir+"/diagnostics.nc"
+
+        pointdata_filename = self.dumpdir+"/point_data.nc"
```

Comment:
It's unclear to me why some of these directories live in `self.output`, others live in `self` (i.e. State) and so on.  So I haven't touched this here.